back/engines/commercial/insights/app/finders/insights/inputs_finder.rb

```diff
@@ -45,6 +51,22 @@ def filter_category(inputs)
       end
     end
 
+    def filter_keywords(inputs)
+      return inputs if params[:keywords].blank?
+
+      networks = view.text_networks.index_by(&:language)
+      keyword_ids = params[:keywords]
+
+      query_terms = keyword_ids.flat_map do |node_id|
+        namespace, _slash, node_id = node_id.partition('/')
+        node = networks[namespace].node(node_id) # raise an exception if the node doesn't exist
+        node.name
+      end
+
+      query = query_terms.uniq.sort.join(' ')
+      inputs.search_any_word(query)
```

Comment:
Weren't the keywords also attached to inputs that contained words close in meaning ? If so I guess this is temporary ?

Reply:
Yes and no 🙂 Words with similar meanings are not grouped into a single keyword. But indeed, inputs can contain different variants of the keyword (inflection / derivation). And yes, those variants will be queried as well in a future iteration.